lib/libc/stdio/lib_tempnam.c

```diff
@@ -103,21 +103,21 @@
 FAR char *tempnam(FAR const char *dir, FAR const char *pfx)
 {
 	FAR char *path;
-	int ret;
+	char *ret;
 
 	if (!dir) {
 		dir = P_tmpdir;
 	}
 
 	if (!pfx) {
-		asprintf(&path, "%s/XXXXXX.tmp", dir);
+		asprintf(&path, "%s/XXXXXX", dir);
 	} else {
-		asprintf(&path, "%s/%s-XXXXXX.tmp", dir, pfx);
+		asprintf(&path, "%s/%.5sXXXXXX", dir, pfx);
 	}
 
 	if (path) {
 		ret = mktemp(path);
-		if (ret == OK) {
+		if (*ret != '\0') {
```

Comment:
It works. But it is better to go by the return value specification rather than implementation. Implementation can change at any time. 
Also  I saw some of glibc test cases as below,
if (mktemp (temppath) == NULL) ... 
So was not clear about return value handling ..